server/pairing/client.go

```diff
@@ -36,17 +36,37 @@ type BaseClient struct {
 
 // NewBaseClient returns a fully qualified BaseClient from the given ConnectionParams
 func NewBaseClient(c *ConnectionParams) (*BaseClient, error) {
-	u, err := c.URL()
-	if err != nil {
-		return nil, err
+
+	var url *url.URL
```

Comment:
This variable shadows the package name `net/url`, meaning calls to the `net/url` package will be ambiguous after this line. That is why the old variable was named `u`, as this name wouldn't shadow / collide with the the `net/url`

Reply:
indeed. Nice catch, will fix 👌